File: src/main/scala/ff/core/Char.scala
package ff.core
import ff.core.Array_._

import ff.core.ArrayBuilder_._

import ff.core.Bool_._

import ff.core.Char_._

import ff.core.Core_._

import ff.core.FileSystem_._

import ff.core.Int_._

import ff.core.List_._

import ff.core.Log_._

import ff.core.Map_._

import ff.core.Nothing_._

import ff.core.Option_._

import ff.core.Pair_._

import ff.core.Set_._

import ff.core.String_._

import ff.core.System_._

import ff.core.Try_._

import ff.core.Unit_._
object Char_ {

type Char = scala.Char;



def Char_isAsciiLetter(self_ : ff.core.Char_.Char) : ff.core.Bool_.Bool = {
self_.isLetter
}

def Char_isAsciiUpper(self_ : ff.core.Char_.Char) : ff.core.Bool_.Bool = {
self_.isUpper
}

def Char_isAsciiLower(self_ : ff.core.Char_.Char) : ff.core.Bool_.Bool = {
self_.isLower
}

def Char_isAsciiDigit(self_ : ff.core.Char_.Char) : ff.core.Bool_.Bool = {
self_.isDigit
}


}

File: src/main/js/ff/core/Char.mjs
import * as ff_core_Array from "../../ff/core/Array.mjs"

import * as ff_core_ArrayBuilder from "../../ff/core/ArrayBuilder.mjs"

import * as ff_core_Bool from "../../ff/core/Bool.mjs"

import * as ff_core_Char from "../../ff/core/Char.mjs"

import * as ff_core_Core from "../../ff/core/Core.mjs"

import * as ff_core_Duration from "../../ff/core/Duration.mjs"

import * as ff_core_FileSystem from "../../ff/core/FileSystem.mjs"

import * as ff_core_Float from "../../ff/core/Float.mjs"

import * as ff_core_Int from "../../ff/core/Int.mjs"

import * as ff_core_List from "../../ff/core/List.mjs"

import * as ff_core_Log from "../../ff/core/Log.mjs"

import * as ff_core_Map from "../../ff/core/Map.mjs"

import * as ff_core_Nothing from "../../ff/core/Nothing.mjs"

import * as ff_core_Option from "../../ff/core/Option.mjs"

import * as ff_core_Pair from "../../ff/core/Pair.mjs"

import * as ff_core_Set from "../../ff/core/Set.mjs"

import * as ff_core_String from "../../ff/core/String.mjs"

import * as ff_core_System from "../../ff/core/System.mjs"

import * as ff_core_Try from "../../ff/core/Try.mjs"

import * as ff_core_Unit from "../../ff/core/Unit.mjs"

// type Char






export function Char_isAsciiLetter(self_) {
return (ff_core_Char.Char_isAsciiUpper(self_) || ff_core_Char.Char_isAsciiLower(self_))
}

export function Char_isAsciiUpper(self_) {
return ((self_ >= 65) && (self_ <= 90))
}

export function Char_isAsciiLower(self_) {
return ((self_ >= 97) && (self_ <= 122))
}

export function Char_isAsciiDigit(self_) {
return ((self_ >= 48) && (self_ <= 57))
}

export function Char_toInt(self_) {
return self_
}


